foresight/utils/test_wrapper.py

```diff
@@ -0,0 +1,264 @@
+from foresight.constants.constants import ForesightConstants
+from thundra.plugins.config.log_config import LogConfig
+from thundra.wrappers.base_wrapper import BaseWrapper
+import thundra.wrappers.wrapper_utils as wrapper_utils
+from thundra.context.plugin_context import PluginContext
+from thundra.application.global_application_info_provider import GlobalApplicationInfoProvider
+from thundra.context.execution_context_manager import ExecutionContextManager
+from thundra.context.tracing_execution_context_provider import TracingExecutionContextProvider
+from foresight.context import (TestSuiteExecutionContext, TestCaseExecutionContext)
+from foresight.sampler.max_count_aware_sampler import MaxCountAwareSampler 
+from thundra.config.config_provider import ConfigProvider
+from thundra.config import config_names
+from foresight.constants.constants import ForesightConstants
+import foresight.utils.generic_utils as utils
+import logging
+
+logger = logging.getLogger(__name__)
+
+class TestWrapper(BaseWrapper):
+
+    """
+        Foresight wrapper util class. It keeps all the generic information for execution of test. 
+        Creating test suite and test case, change application info, start and finish trace and invocation plugins.
+    """
+
+    __instance = None
+
+    @staticmethod
+    def get_instance(*args, **kwargs):
+        return TestWrapper.__instance if TestWrapper.__instance else TestWrapper(*args, **kwargs) 
+
+
+    def __init__(self, *, plugin_executor=None, api_key=None, disable_trace=False, disable_metric=True, disable_log=True, opts=None, **ignored):
+        super(TestWrapper, self).__init__(api_key, disable_trace, disable_metric, disable_log, opts)
+        ExecutionContextManager.set_provider(TracingExecutionContextProvider()) #TODO
+        self.application_info_provider = GlobalApplicationInfoProvider()
```

Comment:
how we are managing multiple application info (for ex. test application info and web app application info)?

Reply:
Application info is handled by PluginContext for each web app and test application by using plugins.